internal/backends/validation.go

```diff
@@ -74,3 +74,38 @@ func validateCollectionName(name string) error {
 
 	return nil
 }
+
+// validateIndexes checks that indexes are valid for FerretDB.
+func validateIndexes(exitingIndexes, newIndexes []IndexInfo) error {
+	for _, index := range newIndexes {
+		if index.Name == "" {
+			return NewError(ErrorCodeIndexNameIsEmpty, nil)
+		}
+
+		// Validation needs to be extended (e.g. to check that index names don't contain illegal symbols, see the issue).
```

Comment:
Please remove it completely, as it introduces error codes that are not handled yet, and we don't know if they will be handled at all